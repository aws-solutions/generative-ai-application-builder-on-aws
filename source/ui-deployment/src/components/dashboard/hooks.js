/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/

import { useEffect, useState } from 'react';
import { fetchDeployments } from './deployments';

export function useDeployments(params = {}) {
    const { pageSize, currentPageIndex: clientPageIndex } = params.pagination || {};
    const { sortingDescending, sortingColumn } = params.sorting || {};
    const { filteringText, filteringTokens, filteringOperation } = params.filtering || {};
    const [loading, setLoading] = useState(false);
    const [items, setItems] = useState([]);
    const [totalCount, setTotalCount] = useState(0);
    const [currentPageIndex, setCurrentPageIndex] = useState(clientPageIndex);
    const [pagesCount, setPagesCount] = useState(0);

    useEffect(() => {
        setCurrentPageIndex(clientPageIndex);
    }, [clientPageIndex]);

    useEffect(() => {
        setLoading(true);
        const params = {
            filteringText,
            filteringTokens,
            filteringOperation,
            pageSize,
            currentPageIndex,
            sortingDescending,
            ...(sortingColumn
                ? {
                      sortingColumn: sortingColumn.sortingField
                  }
                : {})
        };
        const callback = ({ items, pagesCount, currentPageIndex }) => {
            setLoading(false);
            setItems(items);
            setPagesCount(pagesCount + 1);
            setCurrentPageIndex(currentPageIndex);
            if (totalCount === 0) {
                setTotalCount(pagesCount > 1 ? pageSize * (pagesCount - 1) : items.length);
            }
        };
        fetchDeployments(params, callback);
    }, [
        pageSize,
        sortingDescending,
        sortingColumn,
        currentPageIndex,
        filteringText,
        filteringTokens,
        filteringOperation
    ]);

    return {
        items,
        loading,
        totalCount,
        pagesCount,
        currentPageIndex
    };
}
