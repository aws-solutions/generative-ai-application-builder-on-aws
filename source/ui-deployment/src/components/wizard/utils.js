/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/

import { DEFAULT_KENDRA_NUMBER_OF_DOCS, DEFAULT_KNOWLEDGE_BASE_TYPE } from '../../utils/constants';
import {
    BEDROCK_MODEL_OPTION_IDX,
    HF_INF_ENDPOINT_OPTION_IDX,
    HF_MODEL_OPTION_IDX,
    MODEL_FAMILY_PROVIDER_OPTIONS
} from './steps-config';

export const getFieldOnChange =
    (fieldType, fieldKey, onChangeFn) =>
    ({ detail: { selectedOption, value, checked } }) => {
        let fieldValue;
        if (fieldType === 'select') {
            fieldValue = selectedOption;
        } else if (fieldType === 'toggle') {
            fieldValue = checked;
        } else {
            fieldValue = value;
        }
        onChangeFn({
            [fieldKey]: fieldValue
        });
    };

export const createDeployRequestPayload = (stepsInfo) => {
    const payload = {
        ...createKnowledgeBaseApiParams(stepsInfo.knowledgeBase),
        ...createLLMParamsApiParams(stepsInfo.model, stepsInfo.knowledgeBase.isRagRequired),
        ...createConversationMemoryApiParams(),
        ...createUseCaseInfoApiParams(stepsInfo.useCase, stepsInfo.model)
    };

    return payload;
};

export const createUseCaseInfoApiParams = (useCaseStepInfo, modelStepInfo) => {
    const params = {
        UseCaseName: useCaseStepInfo.useCaseName,
        UseCaseDescription: useCaseStepInfo.useCaseDescription,
        ...(useCaseStepInfo.defaultUserEmail &&
            useCaseStepInfo.defaultUserEmail !== '' && {
                DefaultUserEmail: useCaseStepInfo.defaultUserEmail
            }),
        KnowledgeBaseType: DEFAULT_KNOWLEDGE_BASE_TYPE
    };

    if (modelStepInfo.modelProvider !== MODEL_FAMILY_PROVIDER_OPTIONS[BEDROCK_MODEL_OPTION_IDX]) {
        params.ConsentToDataLeavingAWS = true;
    }
    return params;
};

export const stringListToSelectionOptions = (list) => {
    return list.map((item) => {
        return {
            label: item,
            value: item
        };
    });
};

/**
 * Transform the memory step to the params required by the API
 *
 * @param {Object} memoryStepInfo
 */
export const createConversationMemoryApiParams = (memoryStepInfo) => {
    return {
        ConversationMemoryType: 'DynamoDB'
    };
};

/**
 * Transform the model config step to the params required by the API
 * @param {Object} modelStepInfo
 * @returns
 */
export const createLLMParamsApiParams = (modelStepInfo, isRagEnabled = true) => {
    const formatValue = (value, type) => {
        let newValue = value;
        if (type === 'boolean') {
            newValue = newValue.toLowerCase();
        } else if (type === 'list') {
            if (newValue[0] !== '[') {
                newValue = '[' + newValue;
            }
            if (newValue[newValue.length - 1] !== ']') {
                newValue = newValue + ']';
            }
        }
        return newValue;
    };

    const modelParamsObjectCreator = (modelParameters) => {
        if (!modelParameters || modelParameters.length === 0) {
            return {};
        }

        const modelParamsObject = {};
        modelParameters.forEach((modelParam) => {
            modelParamsObject[modelParam.key] = {
                Value: formatValue(modelParam.value, modelParam.type.value),
                Type: modelParam.type.value
            };
        });

        return modelParamsObject;
    };

    const llmParamsPayload = {
        Streaming: modelStepInfo.streaming,
        PromptTemplate: modelStepInfo.promptTemplate,
        Verbose: modelStepInfo.verbose,
        ModelParams: modelParamsObjectCreator(modelStepInfo.modelParameters),
        Temperature: parseFloat(modelStepInfo.temperature),
        RAGEnabled: isRagEnabled
    };

    if (modelStepInfo.modelProvider !== MODEL_FAMILY_PROVIDER_OPTIONS[BEDROCK_MODEL_OPTION_IDX]) {
        llmParamsPayload.ApiKey = modelStepInfo.apiKey;
    }

    if (modelStepInfo.modelProvider === MODEL_FAMILY_PROVIDER_OPTIONS[HF_INF_ENDPOINT_OPTION_IDX]) {
        llmParamsPayload.ModelProvider = MODEL_FAMILY_PROVIDER_OPTIONS[HF_MODEL_OPTION_IDX].value;
        llmParamsPayload.InferenceEndpoint = modelStepInfo.inferenceEndpoint;
    } else {
        llmParamsPayload.ModelProvider = modelStepInfo.modelProvider.value;
        llmParamsPayload.ModelId = modelStepInfo.modelName;
    }

    const payload = {
        LlmParams: llmParamsPayload
    };
    return payload;
};

/**
 * Transform the knowledge base step to the params required by the API. The parameters
 * will be different based on whether the knowledge base step is using an existing index or not.
 * If RAG requirement is set to false in the knowledgeBase step then an empty object is returned
 * @param {Object} knowledgeBaseStep KnowledgeBase step form details
 */
export const createKnowledgeBaseApiParams = (knowledgeBaseStepInfo) => {
    if (!knowledgeBaseStepInfo.isRagRequired) {
        return {};
    }

    if (knowledgeBaseStepInfo.existingKendraIndex === 'yes') {
        // payload for using existing idx
        return {
            KnowledgeBaseParams: {
                ExistingKendraIndexId: knowledgeBaseStepInfo.kendraIndexId,
                NumberOfDocs: knowledgeBaseStepInfo.maxNumDocs ? parseInt(knowledgeBaseStepInfo.maxNumDocs) : 10,
                ReturnSourceDocs: false
            }
        };
    }

    return {
        KnowledgeBaseParams: {
            KendraIndexName: knowledgeBaseStepInfo.kendraIndexName,
            QueryCapacityUnits: knowledgeBaseStepInfo.kendraAdditionalQueryCapacity
                ? parseInt(knowledgeBaseStepInfo.kendraAdditionalQueryCapacity)
                : 0,
            StorageCapacityUnits: knowledgeBaseStepInfo.kendraAdditionalStorageCapacity
                ? parseInt(knowledgeBaseStepInfo.kendraAdditionalStorageCapacity)
                : 0,
            KendraIndexEdition: `${knowledgeBaseStepInfo.kendraEdition.value.toUpperCase()}_EDITION`,
            NumberOfDocs: knowledgeBaseStepInfo.maxNumDocs
                ? parseInt(knowledgeBaseStepInfo.maxNumDocs)
                : DEFAULT_KENDRA_NUMBER_OF_DOCS,
            ReturnSourceDocs: false
        }
    };
};

/**
 * Utility function to update the number of fields in error in entire form based on
 * the error message for a given form field.
 * @param {string} currErrorMessage
 * @param {string} errorState
 * @param {()=>void} setNumFieldsInError
 */
export const updateNumFieldsInError = (currErrorMessage, errorState, setNumFieldsInError) => {
    if (currErrorMessage.length > 0 && errorState.length === 0) {
        setNumFieldsInError((numFieldsInError) => numFieldsInError + 1);
    } else if (currErrorMessage.length === 0 && errorState.length > 0) {
        setNumFieldsInError((numFieldsInError) => numFieldsInError - 1);
    }
};
