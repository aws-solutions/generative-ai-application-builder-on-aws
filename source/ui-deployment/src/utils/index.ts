// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

export * from './test-utils';
export * from './utils';
export * from './dateUtils';
export * from './displayUtils';
export * from './linkUtils';
export * from './tools-content';
export * from './ValueWithLabel';
