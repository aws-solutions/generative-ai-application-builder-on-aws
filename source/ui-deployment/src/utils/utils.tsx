// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import { Auth } from 'aws-amplify';

export interface TransformedDeployment {
    useCaseUUID: string;
    defaultUserEmail: string;
    IsInternalUser?: string;
    createNewVpc: string;
    UseCaseId?: string;
    CreatedDate?: string;
    StackId?: string;
    status?: string;
    UseCaseType?: string;
    UseCaseName?: string;
    Name?: string;
    Description?: string;
    ragEnabled?: string;
    deployUI?: string;
    vpcEnabled?: string;
    vpcId?: string;
    cloudwatchDashboardUrl?: string;
    cloudFrontWebUrl?: string;
    knowledgeBaseType?: string;
    kendraIndexId?: string;
    bedrockKnowledgeBaseId?: string;
    privateSubnetIds?: string[];
    securityGroupIds?: string[];
    KnowledgeBaseParams?: any;
    ConversationMemoryParams?: any;
    LlmParams?: any;
    FeedbackParams?: any;
    AuthenticationParams?: any;
    AgentParams?: any;
    [key: string]: any; // Allow for additional properties
}

/**
 * Use currently authenticated user to generate a JWT token from
 * Cognito.
 * @returns JWT Token
 */
export async function generateToken() {
    try {
        const user = await Auth.currentAuthenticatedUser();
        const token = user.getSignInUserSession().getAccessToken().getJwtToken();
        return token;
    } catch (error) {
        console.error('error REST API:', error);
    }
}

// Copy properties from API response, preserving structure
export const copyIfDefined = (source: any, target: any, sourceKey: string, targetKey?: string) => {
    if (source[sourceKey] !== undefined) {
        target[targetKey || sourceKey] = source[sourceKey];
    }
};

/**
 * Transforms the API response format to match the existing selectedDeployment format
 * preserving the original structure without adding undefined attributes
 * @param apiResponse The response from the getUseCaseDetails API
 * @returns An object matching the format of the existing selectedDeployment
 */
export const mapApiResponseToSelectedDeployment = (apiResponse: any): TransformedDeployment | null => {
    if (!apiResponse) {
        return null;
    }

    // Extract the UUID from the UseCaseId
    const useCaseUUID = apiResponse.UseCaseId ? apiResponse.UseCaseId.substring(0, 8) : '';

    // Create base object with required wizard fields
    const transformedDeployment: TransformedDeployment = {
        useCaseUUID,
        defaultUserEmail: 'placeholder@example.com',
        createNewVpc: 'No'
    };

    // Copy all fields from the API response
    const fields = [
        'UseCaseId',
        'CreatedDate',
        'StackId',
        'UseCaseType',
        'Description',
        'ragEnabled',
        'deployUI',
        'vpcEnabled',
        'vpcId',
        'cloudwatchDashboardUrl',
        'cloudFrontWebUrl',
        'knowledgeBaseType',
        'kendraIndexId',
        'bedrockKnowledgeBaseId',
        'privateSubnetIds',
        'securityGroupIds',
        'KnowledgeBaseParams',
        'ConversationMemoryParams',
        'LlmParams',
        'FeedbackParams',
        'AuthenticationParams',
        'AgentParams'
    ];

    fields.forEach((field) => copyIfDefined(apiResponse, transformedDeployment, field));

    // Special case for Status -> status
    copyIfDefined(apiResponse, transformedDeployment, 'Status', 'status');

    // Special case for UseCaseName -> both UseCaseName and Name
    if (apiResponse.UseCaseName !== undefined) {
        transformedDeployment.UseCaseName = apiResponse.UseCaseName;
        transformedDeployment.Name = apiResponse.UseCaseName;
    }

    // Special case for Bedrock knowledge base ID from nested structure
    if (
        apiResponse.bedrockKnowledgeBaseId === undefined &&
        apiResponse.KnowledgeBaseParams?.BedrockKnowledgeBaseParams?.BedrockKnowledgeBaseId !== undefined
    ) {
        transformedDeployment.bedrockKnowledgeBaseId =
            apiResponse.KnowledgeBaseParams.BedrockKnowledgeBaseParams.BedrockKnowledgeBaseId;
    }

    return transformedDeployment;
};
