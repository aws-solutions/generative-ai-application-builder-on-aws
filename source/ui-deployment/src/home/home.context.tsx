/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/

import { Dispatch, createContext, useEffect } from 'react';

import { ActionType } from '../hooks/useCreateReducer';

import { HomeInitialState } from './home.state';

export interface HomeContextProps {
    state: HomeInitialState;
    dispatch: Dispatch<ActionType<HomeInitialState>>;
}

const HomeContext = createContext<HomeContextProps>(undefined!);

export const HomeContextProvider = (props: any) => {
    useEffect(() => {
        sessionStorage.setItem('init-state', JSON.stringify(props.value.state));
    }, [props.value.state]);

    return <HomeContext.Provider value={props.value}>{props.children}</HomeContext.Provider>;
};

export default HomeContext;
