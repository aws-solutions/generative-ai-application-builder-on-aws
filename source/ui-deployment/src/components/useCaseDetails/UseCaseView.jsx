/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/

import { createRef, useState, useContext } from 'react';
import { useNavigate } from 'react-router-dom';

import { AppLayout, Button, Container, ContentLayout, Header, SpaceBetween, Tabs } from '@cloudscape-design/components';

import { Breadcrumbs, GeneralConfig, PageHeader, ModelDetails, KnowledgeBaseDetails } from './common-components';
import { PromptDetails } from './PromptDetails';
import { Navigation, InfoLink, Notifications } from '../commons/common-components';
import { appLayoutAriaLabels } from '../../i18n-strings';
import { ToolsContent } from './tools-content';
import HomeContext from '../../contexts/home.context';
import { parseStackName } from '../commons/table-config';
import { DeleteDeploymentModal, onDeleteConfirm } from '../commons/delete-modal';
import { CFN_STACK_STATUS_INDICATOR, DEPLOYMENT_ACTIONS } from '../../utils/constants';
import { statusIndicatorTypeSelector } from '../dashboard/deployments';

const Model = ({ loadHelpPanelContent }) => (
    <Container
        header={
            <Header
                variant="h2"
                info={
                    <InfoLink
                        onFollow={() => loadHelpPanelContent(1)}
                        ariaLabel={'Information about deployment model.'}
                    />
                }
            >
                Model
            </Header>
        }
    >
        <ModelDetails isInProgress={true} />
    </Container>
);

const KnowledgeBase = ({ loadHelpPanelContent }) => (
    <Container
        header={
            <Header
                variant="h2"
                info={
                    <InfoLink
                        onFollow={() => loadHelpPanelContent(1)}
                        ariaLabel={'Information about deployment knowledge base.'}
                    />
                }
            >
                Knowledge base
            </Header>
        }
    >
        <KnowledgeBaseDetails isInProgress={true} />
    </Container>
);

const Prompt = ({ loadHelpPanelContent, selectedDeployment }) => (
    <Container
        header={
            <Header
                variant="h2"
                info={
                    <InfoLink
                        onFollow={() => loadHelpPanelContent(1)}
                        ariaLabel={'Information about deployment prompt.'}
                    />
                }
            >
                Prompt
            </Header>
        }
    >
        <PromptDetails selectedDeployment={selectedDeployment} />
    </Container>
);

export default function UseCaseView() {
    const {
        state: { selectedDeployment },
        dispatch: homeDispatch
    } = useContext(HomeContext);
    const navigate = useNavigate();

    const appLayout = createRef();
    const [toolsOpen, setToolsOpen] = useState(false);
    const [, setToolsIndex] = useState(false);
    const [showDeleteModal, setShowDeleteModal] = useState(false);

    const onDeleteInit = () => setShowDeleteModal(true);
    const onDeleteDiscard = () => setShowDeleteModal(false);

    function loadHelpPanelContent(index) {
        setToolsIndex(index);
        setToolsOpen(true);
        appLayout.current?.focusToolsClose();
    }

    const tabs = [
        {
            label: 'Model',
            id: 'model',
            content: <Model loadHelpPanelContent={loadHelpPanelContent} />,
            key: 'model'
        },
        {
            label: 'Knowledge base',
            id: 'knowledgeBase',
            content: <KnowledgeBase loadHelpPanelContent={loadHelpPanelContent} />,
            key: 'knowledgeBase'
        },
        {
            label: 'Prompt',
            id: 'prompt',
            content: <Prompt loadHelpPanelContent={loadHelpPanelContent} selectedDeployment={selectedDeployment} />,
            key: 'prompt'
        }
    ];

    const onEditClickAction = () => {
        homeDispatch({
            field: 'selectedDeployment',
            value: selectedDeployment
        });
        homeDispatch({
            field: 'deploymentAction',
            value: DEPLOYMENT_ACTIONS.EDIT
        });
        navigate(`/wizardView`);
    };

    const onCloneClickAction = () => {
        homeDispatch({
            field: 'selectedDeployment',
            value: selectedDeployment
        });
        homeDispatch({
            field: 'deploymentAction',
            value: DEPLOYMENT_ACTIONS.CLONE
        });
        navigate(`/wizardView`);
    };

    const onFollowNavigationHandler = (event) => {
        navigate(event.detail.href);
    };

    const currentDeploymentStatus = statusIndicatorTypeSelector(selectedDeployment.status);
    const isEditEnabled =
        currentDeploymentStatus === CFN_STACK_STATUS_INDICATOR.SUCCESS ||
        currentDeploymentStatus === CFN_STACK_STATUS_INDICATOR.WARNING;

    return (
        <AppLayout
            ref={appLayout}
            content={
                <ContentLayout
                    header={
                        <PageHeader
                            buttonsList={[
                                <Button
                                    onClick={onEditClickAction}
                                    key={'edit-button'}
                                    data-testid="use-case-view-edit-btn"
                                    disabled={!isEditEnabled}
                                >
                                    Edit
                                </Button>,
                                <Button
                                    onClick={onCloneClickAction}
                                    key={'clone-button'}
                                    data-testid="use-case-view-clone-btn"
                                >
                                    Clone
                                </Button>,
                                <Button onClick={onDeleteInit} key={'delete-button'}>
                                    Delete
                                </Button>
                            ]}
                            deploymentId={parseStackName(selectedDeployment.StackId)}
                        />
                    }
                >
                    <SpaceBetween size="l">
                        <GeneralConfig />
                        <Tabs tabs={tabs} ariaLabel="Resource details" />
                    </SpaceBetween>
                    <DeleteDeploymentModal
                        visible={showDeleteModal}
                        onDiscard={onDeleteDiscard}
                        onDelete={onDeleteConfirm}
                        deployment={selectedDeployment}
                    />
                </ContentLayout>
            }
            breadcrumbs={<Breadcrumbs deploymentId={parseStackName(selectedDeployment.StackId)} />}
            navigation={<Navigation onFollowHandler={onFollowNavigationHandler} />}
            tools={<ToolsContent />}
            toolsOpen={toolsOpen}
            onToolsChange={({ detail }) => setToolsOpen(detail.open)}
            ariaLabels={appLayoutAriaLabels}
            notifications={<Notifications />}
            data-testid="use-case-view"
        />
    );
}
