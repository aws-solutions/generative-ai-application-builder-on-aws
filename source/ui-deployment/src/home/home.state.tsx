/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/

import { DEPLOYMENT_ACTIONS } from '../utils/constants';

export interface HomeInitialState {
    authorized: boolean;
    selectedDeployment: any;
    deploymentsData: any;
    deploymentAction: string;
    runtimeConfig?: any;
    reloadData?: boolean;
}

export const initialState: HomeInitialState = {
    authorized: true,
    selectedDeployment: {},
    deploymentsData: [],
    deploymentAction: DEPLOYMENT_ACTIONS.CREATE,
    reloadData: false
};

export const insertRuntimeConfig = (state: Partial<HomeInitialState>, runtimeConfig: any) => {
    return {
        ...state,
        runtimeConfig
    };
};
