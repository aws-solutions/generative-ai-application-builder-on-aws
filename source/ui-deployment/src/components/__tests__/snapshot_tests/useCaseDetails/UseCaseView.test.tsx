// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import '@testing-library/jest-dom';
import { Dispatch } from 'react';
import { ActionType } from '../../../../hooks/useCreateReducer';
import { HomeInitialState } from '../../../../contexts/home.state';
import UseCaseView from '../../../useCaseDetails/UseCaseView';

// eslint-disable-next-line jest/no-mocks-import
import mockContext from '../../__mocks__/mock-context.json';
import { snapshotWithProvider } from '@/utils';

vi.mock('@cloudscape-design/components');

const contextValue = {
    dispatch: jest.fn() as Dispatch<ActionType<HomeInitialState>>,
    state: mockContext
};

test('Snapshot test', async () => {
    const tree = snapshotWithProvider(<UseCaseView />, '/deployment-details', contextValue).toJSON();
    expect(tree).toMatchSnapshot();
});
