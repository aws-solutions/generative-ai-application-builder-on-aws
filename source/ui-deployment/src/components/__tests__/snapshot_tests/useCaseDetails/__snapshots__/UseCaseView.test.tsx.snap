// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot test 1`] = `null`;
